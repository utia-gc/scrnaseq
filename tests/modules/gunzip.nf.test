nextflow_process {

    name "Test Process gunzip"
    script "modules/gunzip.nf"
    process "gunzip"

    test("gunzip decompresses GZIP-compressed fasta file.") {
        // instantiate helper objects
        def refs = new RefsEnsembl()
>>>>>>> 36485fb849a649865b44b57f024b897ffb619a0c

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
<<<<<<< HEAD
                input[0] = file("https://github.com/utia-gc/ngs-test/raw/scrnaseq/data/references/GRCh38.p14_genomic_NC_000021.9.gtf.gz")
=======
                input[0] = file("${projectDir}/${refs.getGenomeGzip()}")
>>>>>>> 36485fb849a649865b44b57f024b897ffb619a0c
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test unzipped genome fasta
            with(process.out.gunzip) {
                assert size() == 1
<<<<<<< HEAD
                assert get(0) ==~ /^.*\/GRCh38.p14_genomic_NC_000021.9.gtf/
=======
                assert get(0) ==~ /^.*\/genome_I.fa/
>>>>>>> 36485fb849a649865b44b57f024b897ffb619a0c
            }

            // test reproducibility
            assert snapshot(process.out).match()
        }

    }

}
