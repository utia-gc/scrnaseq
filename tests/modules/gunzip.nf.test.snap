{
    "gunzip decompresses GZIP-compressed fasta file.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "gunzip": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
<<<<<<< HEAD
        "timestamp": "2023-08-22T16:22:52+0000"
    },
    "gzip successfully unzips GZIP-compressed fasta file.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "gunzip": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
        "timestamp": "2023-08-22T16:22:52+0000"
    },
    "gunzip successfully unzips GZIP-compressed GTF file.": {
        "content": [
            {
                "0": [
                    "GRCh38.p14_genomic_NC_000021.9.gtf:md5,5dbd6627231b875a9e62136f8dfe2923"
                ],
                "gunzip": [
                    "GRCh38.p14_genomic_NC_000021.9.gtf:md5,5dbd6627231b875a9e62136f8dfe2923"
                ]
            }
        ],
        "timestamp": "2023-08-22T16:22:52+0000"
=======
        "timestamp": "2023-09-13T19:31:52+0000"
>>>>>>> 36485fb849a649865b44b57f024b897ffb619a0c
    }
}